subsys/net/ip/Kconfig.ipv4

```diff
@@ -63,6 +71,13 @@ config NET_DEBUG_DHCPV4
 	help
 	  Enable debug diagnostic from DHCPV4 client.
 
+config NET_DEBUG_IPV4_AUTOCONF
+	bool "Debug IPv4 autoconf client"
+	depends on NET_IPV4_AUTOCONF
```

Comment:
Noticed that I cannot enable debugging, the reason was that the depends on statement is wrong here, it should be "depends on NET_IPV4_AUTO"